crates/cli/src/args.rs

```diff
@@ -0,0 +1,42 @@
+//! This file contains the structure of command line arguments supported by
+//! hieratika-cli.
+
+use std::path::PathBuf;
+
+use clap::{command, Args, Parser, Subcommand, ValueEnum};
+
+#[derive(Clone, Debug, Parser)]
+#[command(
+    version,
+    about,
+    before_help = "Hieratika enables compilation of LLVM bytecode to run in the Cairo VM."
+)]
+pub struct Arguments {
+    /// The wrapper of all the commands available.
+    #[clap(subcommand)]
+    pub command: Command,
+}
+
+#[derive(Clone, Debug, Subcommand)]
+pub enum Command {
+    /// Use Hieratika compiler to generate `FlatLowered`.
+    Build(BuildArgs),
+}
+
+/// The arguments required for the build command.
+#[derive(Args, Clone, Debug)]
+pub struct BuildArgs {
+    /// The compiler to use.
+    #[arg(value_enum)]
+    pub compiler_type: CompilerType,
+
+    /// The project folder or file to compile.
+    pub path: PathBuf,
```

Comment:
```suggestion
    /// The path to the project or file to compile.
    pub path: PathBuf,
```

This should also specify a metavar `PATH`.

Reply:
Do you mean getting the project path from the environment variable? In that case, I would rename the flag to `project_path` because `PATH` is not an available name.

Reply:
No, in clap you can specify a metavar to appear in the help text, which I suppose actually defaults to `PATH` here. Ignore the noise.